metasyn/filehandler.py

```diff
@@ -0,0 +1,191 @@
+import json
+import warnings
+from abc import ABC, abstractmethod
+from pathlib import Path
+from typing import Optional
+
+import polars as pl
+
+_AVAILABLE_FILE_HANDLERS = {}
+
+
+def filehandler(*args):
+    """Register a dataset so that it can be found by name."""
+
+    def _wrap(cls):
+        _AVAILABLE_FILE_HANDLERS[cls.name] = cls
+        return cls
+
+    return _wrap(*args)
+
+class BaseFileHandler(ABC):
```

Comment:
I'm happy with filehandler as internal name, but any user-facing function and message should be clearer w.r.t. functionality: file_format, file_metadata, or so